src/figuresClasses.ts

```diff
@@ -1,19 +1,96 @@
-export interface Figure {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+    ) {
+      throw new Error('Try to put another side length');
+    }
+
+    if (Math.max(this.a, this.b, this.c)
+      >= (this.a + this.b + this.c - Math.max(this.a, this.b, this.c))) {
+      throw new Error('One of your sides is bigger then sum of two other');
+    }
```

Comment:
Don't repeat yourself!
```suggestion
    const maxSideLength = Math.max(this.a, this.b, this.c);
    
    if (maxSideLength >= (this.a + this.b + this.c - maxSideLength)) {
      throw new Error('One of your sides is bigger then sum of two other');
    }
```